docs/problem_format/problem_format.md

```diff
@@ -49,4 +55,16 @@ test_cases:
   batched:
   - {in: tle16p4.2.in, out: tle16p4.2.out}
   - {in: tle16p4.3.in, out: tle16p4.3.out}
+
 ```
+
+As DMOJ's YAML dialect supports dynamic keys, large `init.yml`s can be programmatically generated.
+See the sample files for examples.
+
+## Specifying Cases with Regexes
+If the test cases follow a similar format, it is possible to specify them with a regex.
+
+The default regex for input files is `^(?=.*?\.in|in).*?(?:(?:^|\W)(?P<batch>\d+)[^\d\s]+)?(?P<case>\d+)[^\d\s]*$`, and the default regex for output files is `^(?=.*?\.out|out).*?(?:(?:^|\W)(?P<batch>\d+)[^\d\s]+)?(?P<case>\d+)[^\d\s]*`.
```

Comment:
I can barely read that regex, so providing some examples of which cases it can match is useful.